protostar/self/protostar_compatibility_with_project_checker_test.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+
+from .protostar_compatibility_with_project_checker import (
+    CompatibilityCheckResult,
+    DeclaredProtostarVersionProviderProtocol,
+    ProtostarCompatibilityWithProjectChecker,
+    ProtostarVersionProviderProtocol,
+    parse_protostar_version,
+)
+
+
+class DeclaredProtostarVersionProviderDouble(DeclaredProtostarVersionProviderProtocol):
+    def __init__(self, declared_protostar_version_str: str):
+        self._declared_protostar_version_str = declared_protostar_version_str
+
+    def get_declared_protostar_version(self):
+        return parse_protostar_version(self._declared_protostar_version_str)
+
+
+class ProtostarVersionProviderDouble(ProtostarVersionProviderProtocol):
+    def __init__(self, protostar_version_str: str):
+        self._protostar_version_str = protostar_version_str
+
+    def get_protostar_version(self):
+        return parse_protostar_version(self._protostar_version_str)
+
+
+@pytest.fixture(name="declared_protostar_version")
+def declared_protostar_version_fixture() -> str:
+    assert False, "Not parametrized"
```

Comment:
what's the purpose of this fixture? wouldn't pytest raise anyway on parameter mismatch?